name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/wizards/datatransfer/ImportExtractedItemsWizard.java

```diff
@@ -45,40 +58,95 @@ public Image getDefaultPageImage()
+                    }
+
+                    break;
+
+                // fall back if auto importer is not used
+                default:
+                    ImportAssistantPages.add(new ReviewExtractedItemsPage(client, extractor, preferences, files, "extracted"));
+                    break;
+            }
+            if (ImportAssistantPages.size() > 0) {
+                for (int PDFAssistantPageID = 0; PDFAssistantPageID < ImportAssistantPages.size(); PDFAssistantPageID++) addPage(ImportAssistantPages.get(PDFAssistantPageID));
+            }
+            AbstractWizardPage.attachPageListenerTo(getContainer());
+        }
+        catch (IOException ex)
+        {
+        }
+
     }
 
     @Override
     public boolean performFinish()
     {
-        page.afterPage();
-
-        InsertAction action = new InsertAction(client);
-        action.setConvertBuySellToDelivery(page.doConvertToDelivery());
 
-        boolean isDirty = false;
-        for (ExtractedEntry entry : page.getEntries())
+        if (ImportAssistantPages.size() > 0)
         {
-            if (entry.isImported())
+            boolean isDirty = false;
+            for (int PDFAssistantPageID = 0; PDFAssistantPageID < ImportAssistantPages.size(); PDFAssistantPageID++)
             {
-                entry.getItem().apply(action, page);
-                isDirty = true;
+
+                ImportAssistantPages.get(PDFAssistantPageID).afterPage();
+                InsertAction action = new InsertAction(client);
+                action.setConvertBuySellToDelivery(ImportAssistantPages.get(PDFAssistantPageID).doConvertToDelivery());
+
+                for (ExtractedEntry entry : ImportAssistantPages.get(PDFAssistantPageID).getEntries())
+                {
+                    if (entry.isImported())
+                    {
+                        entry.getItem().apply(action, ImportAssistantPages.get(PDFAssistantPageID));
+                        isDirty = true;
+                    }
+                }
             }
-        }
 
-        if (isDirty)
-        {
-            client.markDirty();
+            if (isDirty)
+            {
+                client.markDirty();
 
-            // run consistency checks in case bogus transactions have been
-            // created (say: an outbound delivery of a security where there no
-            // held shares)
-            new ConsistencyChecksJob(client, false).schedule();
+                // run consistency checks in case bogus transactions have been
+                // created (say: an outbound delivery of a security where there no
+                // held shares)
+                new ConsistencyChecksJob(client, false).schedule();
+            }
         }
 
         return true;
     }
 
+    private List<File> ImportDetectedBankIdentierFiles(List<String> Dateinamen)
```

Comment:
Nicht groß: ich versuche den Code komplett in Englisch zu halten. Es gibt ja schon eine Englische Version - ich mache da aber nicht groß Werbung mit.

Außerdem ist es "naming convention" die Variablen immer mit einem kleinen Buchstaben beginnen zu lassen. Dann kann man das einfach von Klassen unterschieden. Also hier lieber "filenames" als "Dateinamen".